File: Examples/FizzBuzz/FizzBuzz.cs
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.0

using System;
using System.Linq;

namespace Examples.FizzBuzz;

public static class FizzBuzz
{
    public static void Main()
    {
        foreach (var number in Enumerable.Range(1, 30))
        {
            Console.WriteLine(number switch
            {
                _ when number % 15 == 0 => "FizzBuzz",
                _ when number % 3 == 0 => "Fizz",
                _ when number % 5 == 0 => "Buzz",
                _ => number.ToString(),
            });
        }
    }
}

File: Examples/FizzBuzz/FizzBuzz.cpp
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.1

#include <iostream>
#include <ranges>

int
main()
{
    for (const int number : std::views::iota(1, 31))
    {
        if (number % 15 == 0)
        {
            std::cout << "FizzBuzz\n";
        }
        else if (number % 3 == 0)
        {
            std::cout << "Fizz\n";
        }
        else if (number % 5 == 0)
        {
            std::cout << "Buzz\n";
        }
        else
        {
            std::cout << number << '\n';
        }
    }
}
